autopush/websocket.py

```diff
@@ -233,50 +295,19 @@ def verifyNuke(self):
             return
 
         # abortConnection still has failed to shut this down one minute later
-        self.metrics.increment("client.error.abortConnection_failed",
-                               tags=self.base_tags)
+        self.ps.metrics.increment("client.error.abortConnection_failed",
+                                  tags=self.base_tags)
 
     @log_exception
     def onConnect(self, request):
         """autobahn onConnect handler for when a connection has started"""
         track_object(self, msg="onConnect Start")
+        self.ps = PushState(settings=self.ap_settings, request=request)
 
         # Setup ourself to handle producing the data
         self.transport.bufferSize = 2 * 1024
```

Comment:
bufferSize for the transport is the point at which it will call the producers pauseProducing method to halt output. So if the kernel buffer is full, and there's 2kb backed up, it will cause pauseProducing until it can empty out the buffers.